fencing/commands.c

```diff
@@ -1212,6 +1212,20 @@ stonith_device_remove(const char *id, gboolean from_cib)
         g_hash_table_remove(device_list, id);
         crm_info("Removed '%s' from the device list (%d active devices)",
                  id, g_hash_table_size(device_list));
+    } if (!device->cib_registered) {
```

Comment:
oops once there was an else in front of that if and I guess this should do the right thing and gives the most readable code as it makes it more obvious that all cases are covered.
Strange that it didn't segfault in my test. But maybe the else just disappeared when I reformatted a line that was to long before commit.